docs/source/userdoc/requirements/interns/semin-murtic/GPIO_requirements.rst

```diff
@@ -0,0 +1,101 @@
+.. _GPIO_requirements:
+
+GPIO requirements
+=================
+
+.. req:: label
+   :id: REQ_INT3_1732147249
+   :status: Draft
+   :date-released:
+   :priority: Low
+   :submitted-by: Semin Murtic
+   :modified-by:
+   :category: Informational
+   :safety-asil: 
+   :references: 
+   :verification-and-validation: 
+
+- There is a need to understand the structure of STM32F411's GPIO system, which includes multiple ports (GPIOA, GPIOB, etc.), each with 16 pins (0–15).
+- Each pin needs to be configured individually to operate in input, output, alternate function, or analog mode, depending on the application requirements.
+- The GPIO ports need their clocks to be enabled through the RCC->AHB1ENR register before any configuration or operation.
+- Without enabling the clock, no access to the GPIO configuration registers is possible.
+- There is a need to use the MODER register to define the mode of operation for each pin.
+- Modes need to be chosen based on the use case: input (00), output (01), alternate function (10), or analog (11).
+- The OTYPER register needs to be configured to determine whether a pin operates in push-pull or open-drain mode.
+- Push-pull is needed for standard output signals, while open-drain is needed for interfaces like I2C.
+- The OSPEEDR register needs to be configured to set the output speed of a pin.
+- Speed settings need to align with the application's timing requirements and power considerations, with options for low, medium, high, and very high speeds.
```

Comment:
need to align with the application's timing requirements and power considerations - ovo je nepotrebno, samo napisi koji vrste speeda postoje